torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -47,27 +52,45 @@ std::shared_ptr<Operator> matchBuiltinOp(
 
 } // namespace
 
-py::object toPyObj(const Message& message) {
-  switch (message.type()) {
+using namespace torch::distributed::autograd;
+
+py::object toPyObjInternal(RpcCommandBase* rpc, MessageType messageType) {
+  switch (messageType) {
     case MessageType::SCRIPT_RET: {
-      ScriptRet ret = ScriptRet::fromMessage(message);
+      auto ret = static_cast<ScriptResp*>(rpc);
       Stack stack;
-      stack.push_back(ret.value());
+      stack.push_back(ret->value());
       return torch::jit::createPyObjectForStack(std::move(stack));
     }
     case MessageType::PYTHON_RET: {
-      return PythonRpcHandler::getInstance().loadPythonUDFResult(message);
+      // TODO: Try to avoid a copy here.
+      auto resp = static_cast<PythonUDFResp*>(rpc);
+      return PythonRpcHandler::getInstance().loadPythonUDFResult(
+          resp->pickledPayload());
     }
-    case MessageType::EXCEPTION: {
-      std::string err(message.payload().begin(), message.payload().end());
-      throw std::runtime_error(err);
+    case MessageType::MESSAGE_WITH_AUTOGRAD_RESP: {
+      auto rpcWithAutograd = static_cast<RpcWithAutograd*>(rpc);
+      const auto& autogradMetadata = rpcWithAutograd->autogradMetadata();
+
+      // Attach 'recv' autograd function.
+      DistAutogradContext* autogradContext = addRecvRpcBackward(
```

Comment:
nit; this `autogradContext` that is returned. doesn't seem like its being used in the rest of the function. Is there a need for it?